maintainers/patches

```diff
@@ -0,0 +1,88 @@
+#!/bin/zsh
+
+PROGRAM=$0
+
+success() {
+  echo "\e[1;32mDone:\e[m $1"
+  exit 0
+}
+
+fail() {
+  echo "\e[1;31mError:\e[m $1"
+  exit 1
+}
+
+help() {
+  cat <<EOF
+Usage: $PROGRAM {apply|save}
+
+  apply       Applies the patches to the Tock submodule.
+              This resets Tock and can always be called.
+
+  save        Saves the Tock submodule to the patches.
+              Should only be called after apply and when all changes have been
+              added to a commit. After saving, you can run ./setup.sh to return
+              to normal state. Otherwise you can continue editing Tock and
+              calling save.
+EOF
+  exit 0
+}
+
+commit() {
+  local message=$1
+  git add .
+  git commit -qm$message
+}
+
+get_root() {
+  git ls-tree HEAD:third_party | sed -n 's/^.* \([^ ]\+\)\ttock$/\1/p'
+}
+
+get_head() {
+  git rev-parse HEAD
+}
+
+apply() {
+  git submodule -q update third_party/tock
+  ( cd third_party/tock
+    git reset -q --hard
+    git clean -qfxd
+    rsync -a ../../boards .
+    commit '00-boards'
+    for file in ../../patches/tock/*; do
```

Comment:
The `setup-submodules` script uses
```
for p in ../../patches/tock/[0-9][0-9]-*.patch; do
```

Might be good to be consistent?

Reply:
I don't think it's beneficial. I prefer this script to be more generic to how files are named such that if we change our nomenclature we don't break / need to update this tool.

Reply:
I don't have a strong preference either way, I just thought it might be a source for confusion if scripts behaves differently. Counter argument would be a case like wanting to store a `README.md`? How about both scripts take all `.patch` files?

Reply:
Oh I see, indeed one could use this script to rename patches in a way that would not work with setup.sh anymore. Good point. I would be more in favor of using all patches because what would the patches not satisfying the format be for? They would be confusing I think.

Reply:
The reason the filenames for patches start with numbers is simply to ensure they are applied in a specific consistant order.

Reply:
Yes we rely on the fact that pathname expansion is in alphabetical order but we don't need the patch names to start with exactly 2 digits and a dash. This may be modified in the future (if needed) and the scripts should still work the same.